packages/app/src/specs/SpecHeaderCloudDataTooltip.vue

```diff
@@ -0,0 +1,204 @@
+            :prefix-icon="SendIcon"
+            prefix-icon-class="icon-dark-white icon-light-transparent"
+            data-cy="access-requested-button"
+            class="bg-gray-800 border-gray-800"
+            disabled
+          >
+            {{ t("specPage.requestSentButton") }}
+          </Button>
+        </div>
+      </div>
+    </template>
+  </Tooltip>
+</template>
+
+<script setup lang="ts">
+import Button from '@cy/components/Button.vue'
+import Tooltip from '@packages/frontend-shared/src/components/Tooltip.vue'
+import ConnectIcon from '~icons/cy/chain-link_x16.svg'
+import UserOutlineIcon from '~icons/cy/user-outline_x16.svg'
+import SendIcon from '~icons/cy/paper-airplane_x16.svg'
+import ExternalLink from '@cy/gql-components/ExternalLink.vue'
+import type { SpecHeaderCloudDataTooltipFragment } from '../generated/graphql'
+import { SpecHeaderCloudDataTooltip_RequestAccessDocument } from '../generated/graphql'
+import { useI18n } from '@cy/i18n'
+import { computed, onMounted, ref } from 'vue'
+import { gql, useMutation } from '@urql/vue'
+const { t } = useI18n()
+
+const emits = defineEmits<{
+  (eventName: 'showLogin'): void
+  (eventName: 'showConnectToProject'): void
+}>()
+
+const props = defineProps<{
+  gql: SpecHeaderCloudDataTooltipFragment
+  headerTextKeyPath: string
+  headerShortTextKeyPath?: string
+  connectedTextKeyPath: string
+  notConnectedTextKeyPath: string
+  noAccessTextKeyPath: string
+  docsUrl: string
+  docsTextKeyPath: string
+}>()
+
+gql`
+fragment SpecHeaderCloudDataTooltip on Query {
+  currentProject {
+    id
+    cloudProject{
+      __typename
+      ... on CloudProjectUnauthorized {
+        hasRequestedAccess
+      }
+    }
+  }
+  ...Auth
+  ...CloudConnectModals
+}
+`
+
+gql`
+mutation SpecHeaderCloudDataTooltip_RequestAccess( $projectId: String! ) {
+  cloudProjectRequestAccess(projectSlug: $projectId) {
+    __typename
+    ... on CloudProjectUnauthorized {
+      message
+      hasRequestedAccess
+    }
+  }
+}
+`
+
+const hasRequestedAccess = ref(false)
+
+onMounted(() => {
+  if (props.gql.currentProject?.cloudProject?.__typename === 'CloudProjectUnauthorized') {
+    hasRequestedAccess.value = props.gql.currentProject.cloudProject.hasRequestedAccess ?? false
+  }
+})
+
+const projectConnectionStatus = computed(() => {
+  if (!props.gql.cloudViewer) return 'LOGGED_OUT'
+
+  if (!props.gql.currentProject?.cloudProject?.__typename) return 'NOT_CONNECTED'
+
+  if (props.gql.currentProject?.cloudProject?.__typename === 'CloudProjectNotFound') return 'NOT_FOUND'
+
+  if (props.gql.currentProject?.cloudProject?.__typename === 'CloudProjectUnauthorized') {
+    if (hasRequestedAccess.value) {
+      return 'ACCESS_REQUESTED'
+    }
+
+    return 'UNAUTHORIZED'
+  }
+
+  return 'CONNECTED'
+})
+
+const requestAccessMutation = useMutation(SpecHeaderCloudDataTooltip_RequestAccessDocument)
+
+async function requestAccess () {
```

Comment:
Rather than a composable, I'd see it being encapsulated in a `RequestAccessButton.vue` or something. I'll defer to @mike-plummer on whether we want to change this though

Reply:
I agree that this is duplicated behavior and we should combine them. To do it right I would want a better understanding around the other comment regarding our manual state management here vs. the GQL mutation flowing the updated state back into the component